src/components/dialogs/AddStaffNotationDialog.vue

```diff
@@ -0,0 +1,236 @@
+<template>
+  <v-dialog v-model="dialog" width="45rem" persistent :attach="attach" content-class="add-notation-dialog">
+    <v-card>
+      <v-card-title id="dialog-title">Add a {{itemName}}</v-card-title>
+      <v-card-text>
+        <div id="notation-text" class="mb-4 mt-2 default-text">
+          Enter a {{itemName}} that will appear on the ledger for this entity
+        </div>
+        <v-form ref="notationFormRef" v-model="notationFormValid" id="notation-form">
+          <v-textarea
+            v-model="notation"
+            class="text-input-field mb-2"
+            filled
+            :label="itemName"
+            id="notation"
+            rows="4"
+            :rules="notationRules"
+            :counter="notationMaxLength"
+          />
+        </v-form>
+        <div class="default-text">
+          If this filing is pursuant to a court order, enter the court order number. If this filing is pursuant
+          to a plan of arrangement, enter the court order number and select Plan of Arrangement.
+        </div>
+        <CourtOrderPoa
+          id="court-order"
+          @emitCourtNumber="setFileNumber($event)"
+          @emitPoa="setHasPlanOfArrangement($event)"
+          :displaySideLabels="false"
+          :key="courtOrderKey"
+          :autoValidation="enableValidation"
+          :courtOrderNumberRequired="courtOrderNumberRequired"
+          ref="courtOrderPoaRef"
+        />
+      </v-card-text>
+      <v-divider class="mb-4"></v-divider>
+      <v-card-actions class="pt-0">
+        <v-spacer></v-spacer>
+        <div class="form__btns">
+          <v-btn text color="primary"
+            id="dialog-save-button"
+            :loading="saving"
+            @click.native="save()"
+          >Save</v-btn>
+          <v-btn text color="secondary"
+            id="dialog-cancel-button"
+            :disabled="saving"
+            :loading="saving"
+            @click.native="emitClose(false)"
+          >Cancel</v-btn>
+        </div>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop, Watch, Emit, Mixins } from 'vue-property-decorator'
+import { DateMixin } from '@/mixins'
+import { mapState } from 'vuex'
+import axios from '@/axios-auth'
+import { CourtOrderPoa } from '@bcrs-shared-components/court-order-poa'
+import { FormIF } from '@/interfaces'
+
+@Component({
+  computed: {
+    ...mapState(['entityIncNo'])
```

Comment:
1. Use a getter instead of reading state.
2. As this is a new TS component, can we use the `@Getter` style instead of this old style?

Reply:
I just followed the other components. Let me check if that works.

Reply:
Well there's some history here: this app originally didn't use the new class style, so the decoration was how you "imported" state, getters and setters. Now this app has a mixture of component styles. Since this is a net-new component, it would be great to use the new style exclusively instead of a hybrid style. If it works :)  (If not then it probably means we have to update a library, which is not in scope, so you could then ignore this comment.)

Reply:
Yes, I tried to import it but can't find `vuex-class` :)

Reply:
What if you add it as a dependency?

Reply:
I added the dependency and create a getter for `entityIncNo`. It works! 👍